include/MemoryPool.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * MemoryPool.h
+ *
+ * Copyright (c) 2018 Lukas W <lukaswhl/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This file is licensed under the MIT license. See LICENSE.MIT.txt file in the
+ * project root for details.
+ *
+ */
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+
+class _MemoryPool_Private;
+
+class _MemoryPool_Base
+{
+public:
+	_MemoryPool_Base(size_t size, size_t nmemb);
+	virtual ~_MemoryPool_Base();
+	void * allocate();
+	void * allocate_bounded();
+	void deallocate(void * ptr);
+
+private:
+	const std::unique_ptr<_MemoryPool_Private> _imp;
+};
+
+/// Thread-safe, lockless memory pool. Only supports allocate(n) with n=1. When
+/// the pool is exhausted, MmAllocator is used.
+template<typename T>
+class MemoryPool : private _MemoryPool_Base
+{
+public:
+	using value_type = T;
+	template<class U> struct rebind { typedef MemoryPool<U> other; };
+
+	MemoryPool(size_t nmemb) :	_MemoryPool_Base(sizeof(T), nmemb) {}
```

Comment:
A space will be better than a tab here.
```suggestion
	MemoryPool(size_t nmemb) : _MemoryPool_Base(sizeof(T), nmemb) {}
```